tls/irc_client_tls.ml

```diff
@@ -0,0 +1,48 @@
+open Lwt.Infix
+
+module Io = struct
+  type 'a t = 'a Lwt.t
+  let (>>=) = Lwt.bind
+  let return = Lwt.return
+
+  type file_descr = {
+    ic: Tls_lwt.ic;
+    oc: Tls_lwt.oc;
+  }
+
+  type inet_addr = string
+
+  type config = Tls.Config.client
+
+  let default_config : Tls.Config.client =
+    Tls.Config.client ~authenticator:X509.Authenticator.null ()
```

Comment:
I am totally not surprised :-)
However, I don't really know what the good default is. I don't know much about TLS and even less about ocaml-tls (there is no easy tutorial or intro beyond the API documentation).

Given that this is about IRC clients, it's unrealistic to pin a particular certificate in advance, so what should be the proper authentication scheme? Just checking that the certificate corresponds to the host name?